### PR TITLE
Add initial metadata-only phase to inconsistent update handling 

### PR DESCRIPTION
**Pull request by vekterli**
@geirst please review. Some notes:
* `TwoPhaseUpdate...` classes and tests should be renamed (since it's more than 2 now), but enough refactoring done in this PR as it is 🙂
* This features is config disabled by default for now

If bucket replicas are inconsistent, the common case is that only
a small subset of documents contained in the buckets are actually
mutually out of sync. The added metadata phase optimizes for
such a case by initially sending Get requests to all divergent
replicas that only ask for the timestamps (and no fields). This
is a very cheap and fast operation. If all returned timestamps are
in sync, the update can be restarted in the fast path. Otherwise, a
full Get will _only_ be sent to the newest replica, and its result will
be used for performing the update on the distributor itself, before
pushing out the result as Puts.

This is in contrast to today's behavior where full Gets are sent
to all replicas. For users with large documents this can be very
expensive.

In addition, the metadata Get operations are sent with weak internal
read consistency (as they do not need to read any previously written,
possibly in-flight fields). This lets them bypass the main commit queue
entirely.

This relates to issue #3703

### CODE REVIEW

**Review by geirst — APPROVED** [review]
👍

File: storage/src/vespa/storage/distributor/operations/external/newest_replica.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <vespa/document/bucket/bucketid.h>
+#include <vespa/storageapi/defs.h>
+#include <iosfwd>
+#include <stddef.h>
+
+namespace storage::distributor {
+
+struct NewestReplica {
```
[the remark this answers is not in the document]
Comment (vekterli) [reply storage/src/vespa/storage/distributor/operations/external/newest_replica.h]: Done

File: storage/src/vespa/storage/distributor/operations/external/newest_replica.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#pragma once
```
[the remark this answers is not in the document]
Comment (vekterli) [reply storage/src/vespa/storage/distributor/operations/external/newest_replica.h]: Done

File: storage/src/vespa/storage/distributor/operations/external/newest_replica.cpp [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#include "newest_replica.h"
```
[the remark this answers is not in the document]
Comment (vekterli) [reply storage/src/vespa/storage/distributor/operations/external/newest_replica.cpp]: Done

**Comment by geirst** [comment]
👍

**Pull request merged**